SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs

```diff
@@ -11,50 +13,66 @@ namespace Microsoft.AppCenter
     /// </summary>
     public partial class AppCenter
     {
-        // Gets the first instance of an app secret corresponding to the given platform name, or returns the string 
+        const string SecretDelimiter = ";";
+        const string PlatformKeyValueDelimiter = "=";
+        const string TargetKeyName = "target";
+        const string AppSecretKeyName = "appsecret";
+
+        // Gets the first instance of an app sceret and/or target token corresponding to the given platform name, or returns the string 
         // as-is if no identifier can be found. Logs a message if no identifiers can be found.
-        internal static string GetSecretForPlatform(string secrets, string platformIdentifier)
+        internal static string GetSecretAndTargetForPlatform(string secrets, string platformIdentifier)
         {
+            var platformTargetIdentifier = platformIdentifier + TargetKeyName;
             if (string.IsNullOrEmpty(secrets))
             {
                 throw new AppCenterException("App secrets string is null or empty");
             }
 
             // If there are no equals signs, then there are no named identifiers, but log a message in case the developer made 
             // a typing error.
-            if (!secrets.Contains("="))
+            if (!secrets.Contains(PlatformKeyValueDelimiter))
             {
                 AppCenterLog.Debug(AppCenterLog.LogTag, "No named identifier found in appSecret; using as-is");
                 return secrets;
             }
 
-            var parseErrorMessage = $"Error parsing key for '{platformIdentifier}'";
+            // Grouping by the name of the key.
+            var secretsGroup = secrets.Split(SecretDelimiter.ToCharArray())
+                                           .Select(value => value.Split(PlatformKeyValueDelimiter.ToCharArray()))
+                                           .GroupBy(p => p[0].Trim() ?? "", StringComparer.OrdinalIgnoreCase);
 
-            var platformIndicator = platformIdentifier + "=";
-            var secretIdx = secrets.IndexOf(platformIndicator, StringComparison.Ordinal);
-            if (secretIdx == -1)
+            // Create a dictionary choosing the last secret value for each key. If the key has more than one secret value then select the last of them.
+            var secretsDictionary = secretsGroup.ToDictionary(pair => pair.Key.Trim(), pair => pair.Last().Last().Trim(), StringComparer.OrdinalIgnoreCase);
+            if (secretsDictionary.ContainsKey(TargetKeyName) || secretsDictionary.ContainsKey(AppSecretKeyName))
             {
-                throw new AppCenterException(parseErrorMessage);
+                AppCenterLog.Debug(AppCenterLog.LogTag, "Found named identifier in the secret; using as-is.");
```

Comment:
I still don't understand this.
If we find named identifiers, we use as is, I don't understand the message being logged.
This needs to be clarified.